builder/metadata.go

```diff
@@ -149,6 +151,48 @@ func (b *Builder) getUpstreamFormatRange(version string) (format string, first,
 	return format, first, latest, err
 }
 
+// ModifyBundles goes through the bundle directory and performs an action when it finds
+// a Deprecated bundle
+func (b *Builder) ModifyBundles(action func(string) error) error {
+	path := b.Config.Mixer.LocalBundleDir
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	var scanner *bufio.Scanner
+	for _, file := range files {
+		fileToScan := filepath.Join(path, file.Name())
+		f, err := os.Open(fileToScan)
+		if err != nil {
+			return err
+		}
+
+		// Scan the files and find which bundle definitions are marked deprecated
+		scanner = bufio.NewScanner(f)
+		var str string
+		re := regexp.MustCompile("#\\s\\[STATUS\\]:\\s*Deprecated.*")
```

Comment:
Also, you should definitely move this to before the for loop. You are compiling the regexp on every iteration yet it doesn't change.